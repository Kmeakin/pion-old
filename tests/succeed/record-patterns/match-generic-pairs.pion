//~ run = "${PION} elab ${FILE}"

let fst = fun (A : Type) (B : Type) (p : (A, B)) => p;
let snd = fun (A : Type) (B : Type) (p : (A, B)) => p;
{}
