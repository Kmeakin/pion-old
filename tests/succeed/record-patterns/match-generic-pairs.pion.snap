exit status: 0
stdout:
{
    let fst : fun (A : Type) (B : Type) -> (A, B) -> (A, B) = fun (A : Type)
    (B : Type)
    (p : (A, B)) => p;
    let snd : fun (A : Type) (B : Type) -> (A, B) -> (A, B) = fun (A : Type)
    (B : Type)
    (p : (A, B)) => p;
    ()
} : ()

stderr:
